llm/server.go

```diff
@@ -527,21 +539,20 @@ func (s *llmServer) getServerStatus(ctx context.Context) (ServerStatus, error) {
 		return ServerStatusError, fmt.Errorf("read health request: %w", err)
 	}
 
-	var status ServerStatusResp
+	var status ServerStatusResponse
 	if err := json.Unmarshal(body, &status); err != nil {
 		return ServerStatusError, fmt.Errorf("health unmarshal encode response: %w", err)
 	}
 
-	switch status.Status {
-	case "ok":
-		return ServerStatusReady, nil
-	case "no slot available":
-		return ServerStatusNoSlotsAvailable, nil
-	case "loading model":
+	result := NewServerStatus(status)
+	switch result {
```

Comment:
So essentially I'd want us to be able to go from `if err := json.Unmarshal(body, &status);...` to `switch status`

Reply:
Good callout, made this change to use the iota directly in the runners. I was preserving too much of the existing pattern here.